src/core/txs/base_tx.c

```diff
@@ -25,6 +25,49 @@ int size_tx(Transaction *tx){
     tx->num_inputs * size_input() + tx->num_outputs * sizeof(Output));
 }
 
+void free_tx(Transaction *tx){
+  if(tx != NULL){
+    if(tx->inputs != NULL){
+      for(unsigned int i = 0; i < tx->num_inputs; i++ ){
+        if(tx->inputs[i].pub_key != NULL){
+          mbedtls_ecp_point_free(tx->inputs[i].pub_key);
+        }
+      }
+      free(tx->inputs);
+    }
+    if(tx->outputs != NULL){
+      free(tx->outputs);
+    }
+  free(tx);
+  }
+}
+
+Transaction *copy_tx(Transaction *tx){
+  Transaction *copy = malloc(sizeof(Transaction));
+
+  // Copy inputs
+  // Harder since we need to also copy pub keys
+  copy->num_inputs = tx->num_inputs;
+  copy->inputs = malloc(tx->num_inputs * sizeof(Input));
+  for(unsigned int i = 0; i<tx->num_inputs; i++){
+    // Copy everything then overwrite the key pointer
+    memcpy(&(copy->inputs[i]), &(tx->inputs[i]), sizeof(Input));
+    copy->inputs[i].pub_key = malloc(sizeof(mbedtls_ecp_point));
```

Comment:
This seems safest, not 100% that we need the additional malloc (From what I saw it didn't look like the context actually had pointers but you are more familiar with the datastructure)